src/Components/Components/test/RendererTest.cs

```diff
@@ -3066,6 +3066,27 @@ public void DisposingRenderer_CapturesExceptionsFromAllRegisteredComponents()
             Assert.Contains(exception2, renderer.HandledExceptions);
         }
 
+        [Fact]
+        public void RendersAreBuffered_WhenCanRenderBatchIsReset()
+        {
+            // Arrange
+            var renderer = new TestRenderer();
+            renderer.SetCanRenderBatch(false);
+            var component = new TestComponent(builder =>
+            {
+                builder.AddContent(0, "Hello");
+                builder.OpenComponent<DisposableComponent>(1);
+                builder.CloseComponent();
+            });
+
+            var componentId = renderer.AssignRootComponentId(component);
+            component.TriggerRender();
+
+            Assert.Empty(renderer.Batches);
+            var entry = Assert.Single(renderer.RenderBatchBuilder.ComponentRenderQueue);
+            Assert.Equal(componentId, entry.ComponentState.ComponentId);
+        }
```

Comment:
This one unit test seems like a very light amount of test coverage for the amount of changes in `Renderer.cs`.

I would have thought there were more new behaviors than this, especially when including the new `protected` extensibility points for subclasses.

I know you mentioned you were reverting some part of the `Renderer.cs` changes (e.g., to remove the protected `Dispatcher`), so don't know how close it's going to go back to its original set of visible behaviors. Perhaps this one test really does cover everything that's going to end up changed.

Reply:
Most of the changes to `Renderer` are removed.